### ISSUE TITLE
Have you considered Gitdown before creating Hercule

### ISSUE THREAD

**Issue by gajus** [jamesramsay/hercule#45]
https://github.com/gajus/gitdown does all of the things as Hercule (transclusion) and more.

This is not meant as an advertisement and more of a "lets combine the forces".

**Comment by zdne** [comment]
interestingly enough Hercule seems to be 8 days older 😜

**Comment by gajus** [comment]
You are right. I have looked at the last commit, number of stargazers and assumed it was released just two months ago.

Wow. Even the original syntax is similar (https://github.com/jamesramsay/hercule/tree/db2bc86e378d89c4f25e075547195fff7576a297).

**Comment by gajus** [comment]
This reminds me https://en.wikipedia.org/wiki/Multiple_discovery

I wonder what triggered us both at the same time to start a similar project.

**Comment by gajus** [comment]
I admittedly like the Hercule transclusion syntax a lot more than Gitdown.

```
:[example link](foo.md)
```

Looks very clean.

That said, Gitdown syntax must support a lot more features and configuration.

**Comment by jamesramsay** [comment]
What a funny coincidence!

Hercule actually began it's life a few years ago as ~25 lines of bash script to stitch text files together, relying on the naming of the files and folders to get them into the right order.

I've got a few local branches where I've started refactoring Hercule to be more configurable an easy to plug into tools like gulp. Maybe I should try and find the time to look into this again...

To your original suggestion of considering combining forces, I'm not against the idea, but would need to understand more practically how you think the projects could work together (Hercule becomes a dependency of Gitdown perhaps?) or become a single project.

More generally, my preference is for small modular tools/packages that can interact with each other nicely. This approach seems to make sense with preprocessors where a use might want to plug a range of them together in varying orders depending on their needs.

I'm interested to hear your thoughts and keep the conversation going, albeit probably at a slowish pace because I've got quite a lot on for the next few weeks.
